### DISCUSSION TITLE
Uvicorn >=0.30 possibly have a worse RAM and CPU management than previous versions

### DISCUSSION THREAD

**Discussion by korcky** [Kludex/uvicorn#2492] [Potential Issue]
It seems like "New multiprocess manager (https://github.com/encode/uvicorn/pull/2183)" introduced in [Version 0.30.0](https://github.com/encode/uvicorn/releases/tag/0.30.0) have a significantly worse performance.

The setup on which I see such an effect is a FastAPI app that runs via uvicorn (8 workers) in a pod in Kubernetes. The uvicorn was updated from `0.29.0` to `0.32.0`, the comparison between CPU and RAM usage can be seen below (solid line is `0.32.0`, doted is `0.29.0`):

<img width="1173" alt="CPU" src="https://github.com/user-attachments/assets/5fbf51ff-e1bf-4c86-a9ad-9e8ba01e8848">
<img width="1173" alt="RAM" src="https://github.com/user-attachments/assets/c44469dc-88a6-4033-99c4-3e5d657a2843">

Besides bumping the version of uvicorn, there were small changes to the service itself and version bumps of other modules but they should be insignificant to the CPU/RAM usage.

And I guess this might be related to this discussion (https://github.com/encode/uvicorn/discussions/2450):
- before upgrading uvicorn, all workers were healthy (**no** child processes were dying)
- after the update we saw the "Child process [...] died" log almost every 10 seconds
- after the memory limits were increased (from `1Gb` to `2GB`), we see the same log once an hour now

Does anyone see a similar behaviour?

**Comment by coolbeevip** [comment]
I ran into an issue with version 0.32.0, which shows "child processes died" on both Windows and Alpine Docker, but it works perfectly on macOS.

In contrast, version 0.29.0 operates normally across Windows, Linux, macOS, and Alpine Docker.
